packages/gatsby-plugin-image/src/components/intersection-observer.ts

```diff
@@ -20,13 +35,18 @@ export function createIntersectionObserver(
       entries => {
         entries.forEach(entry => {
           if (entry.isIntersecting) {
-            callback()
+            // Get the matching entry's callback and call it
+            ioEntryMap.get(entry.target as HTMLElement)?.()
+            // We only need to call it once
+            ioEntryMap.delete(entry.target as HTMLElement)
           }
         })
       },
       {
-        // TODO look at lighthouse & amp-image on what values they are using
-        rootMargin: `150%`,
+        rootMargin:
+          connectionType === `4g`
```

Comment:
could you also check `connection.saveData`
```suggestion
          connectionType === `4g` && !connection.saveData
```